Sources/URITemplate.swift

```diff
@@ -210,7 +210,7 @@ public struct URITemplate : CustomStringConvertible, Equatable, Hashable, Expres
         var extractedVariables:[String: String] = [:]
 
         for (index, variable) in variables.enumerated() {
-          let range = result.rangeAt(index + 1)
+          let range = result.range(at: index + 1)
```

Comment:
Can you make this conditional so that it will compile when used in both Swift 3 and Swift 4 please.